opm/material/densead/Evaluation.hpp

```diff
@@ -295,30 +308,27 @@ class Evaluation
     template <class RhsValueType>
     Evaluation& operator=(const RhsValueType& other)
     {
-        this->value_ = other;
-        std::fill(this->derivatives_.begin(), this->derivatives_.end(), 0.0);
+        setValue( other );
+        clearDerivatives();
         return *this;
     }
 
     // copy assignment from evaluation
     Evaluation& operator=(const Evaluation& other)
     {
-        this->value_ = other.value_;
-        std::copy(other.derivatives_.begin(), other.derivatives_.end(), this->derivatives_.begin());
+        data_ = other.data_;
+
         return *this;
     }
 
     template <class RhsValueType>
     bool operator==(const RhsValueType& other) const
-    { return this->value_ == other; }
+    { return data_[ valuepos_ ] == other; }
 
     bool operator==(const Evaluation& other) const
     {
-        if (this->value_ != other.value_)
-            return false;
-
-        for (unsigned varIdx = 0; varIdx < size; ++varIdx)
-            if (this->derivatives_[varIdx] != other.derivatives_[varIdx])
+        for (int varIdx = 0; varIdx <= size; ++varIdx)
```

Comment:
this is pretty non-obvious since `size` is the number of derivatives. better use `data_.size()` or maybe even `return data_ == other.data_;` works?